### PR TITLE
Update docs about resources watched by the Controller

### PR DESCRIPTION
**Pull request by dacamposol**
**Description of the change:**
Adds explanation and example about how to correctly use Owns in the Controller to watch a secondary dependent resource.

**Motivation for the change:**
https://github.com/operator-framework/operator-sdk/issues/5949

**Checklist**
If the pull request includes user-facing changes, extra documentation is required:
- [ ] Add a new changelog fragment in `changelog/fragments` (see [`changelog/fragments/00-template.yaml`](https://github.com/operator-framework/operator-sdk/tree/master/changelog/fragments/00-template.yaml))
- [x] Add or update relevant sections of the docs website in [`website/content/en/docs`](https://github.com/operator-framework/operator-sdk/tree/master/website/content/en/docs)

### CODE REVIEW

**Review by camilamacedo86 — APPROVED** [review]
I am OK with this one. if you could not figure out and understand how it works probably others will be. I just add a nit suggestion.

File: website/content/en/docs/building-operators/golang/tutorial.md
Comment (camilamacedo86) [inline website/content/en/docs/building-operators/golang/tutorial.md:184]:
```suggestion
The dependent objects, in this case the Deployments, need to have an [Owner References](https://kubernetes.io/docs/concepts/overview/working-with-objects/owners-dependents/#owner-references-in-object-specifications) field that references their owner object. This will be added by using the method `ctrl.SetControllerReference`. [More info][ks8-doc-owner-ref]

Note: The K8s api will manage the resources according to the`ownerRef` which will be properly set by using this method. Therefore, the K8s API will know that these resources created on the controller, such as the Deployment to run the Memcached Operand image, depend on the custom resource for the Memcached Kind which will also allow the K8s API delete all resources that were created when/if the custom resource itself be deleted. [More info][k8s-doc-deleting-cascate]
```

We need to add the links alias at the bottom:
[ks8-doc-owner-ref]: https://kubernetes.io/docs/concepts/overview/working-with-objects/owners-dependents/
[k8s-doc-cascading-deletion]: https://kubernetes.io/docs/concepts/architecture/garbage-collection/#cascading-deletion

Comment (camilamacedo86) [reply website/content/en/docs/building-operators/golang/tutorial.md:184]: /hold 
Until address the nit changes

Comment (dacamposol) [reply website/content/en/docs/building-operators/golang/tutorial.md]: @camilamacedo86 done.

**Comment by dacamposol** [comment]
@camilamacedo86 I think you need to remove the tag`do-not-merge/hold` in order to be able to merge this.

**Comment by camilamacedo86** [comment]
/hold cancel 

Let's see what the others think about

**Comment by dacamposol** [comment]
@rashmigottipati could you please check if this could be merged?

File: website/content/en/docs/building-operators/golang/tutorial.md
Comment (everettraven) [inline website/content/en/docs/building-operators/golang/tutorial.md:186]: Small nit:

I don't see where the `k8s-doc-deleting-cascate` link reference exists at the end of the document. Make sure to add it at the end of the document similar to the other link references or else we will have a dead link.

Also looks like there may be a typo in the link reference name:
```suggestion
Note: The K8s api will manage the resources according to the`ownerRef` which will be properly set by using this method. Therefore, the K8s API will know that these resources created on the controller, such as the Deployment to run the Memcached Operand image, depend on the custom resource for the Memcached Kind which will also allow the K8s API delete all resources that were created when/if the custom resource itself be deleted. [More info][k8s-doc-deleting-cascade]
```

File: website/content/en/docs/building-operators/golang/tutorial.md
Comment (everettraven) [inline website/content/en/docs/building-operators/golang/tutorial.md:184]: Small nit:

I don't see where the `ks8-doc-owner-ref` link reference exists at the end of the document. Make sure to add it at the end of the document similar to the other link references or else we will have a dead link.

Also looks like there may be a typo in the link reference name:
```suggestion
The dependent objects, in this case the Deployments, need to have an [Owner References](https://kubernetes.io/docs/concepts/overview/working-with-objects/owners-dependents/#owner-references-in-object-specifications) field that references their owner object. This will be added by using the method `ctrl.SetControllerReference`. [More info][k8s-doc-owner-ref]
```

**Review by everettraven — CHANGES_REQUESTED** [review]
Hi @dacamposol ! Thanks for the contribution!

I think this is a valuable addition to the tutorial, I just have a couple minor nits that I think need to be addressed before it can be merged.

**Comment by dacamposol** [comment]
> Hi @dacamposol ! Thanks for the contribution!
> 
> I think this is a valuable addition to the tutorial, I just have a couple minor nits that I think need to be addressed before it can be merged.

I have consumed your PR comments.

Please check :)

**Comment by jmrodri** [comment]
@dacamposol also if you could sign your commits that would fix the DCO error. `git rebase HEAD~3 --signoff` or however many commits were not signed.

**Comment by dacamposol** [comment]
> @dacamposol also if you could sign your commits that would fix the DCO error. `git rebase HEAD~3 --signoff` or however many commits were not signed.

You cannot rebase a merge commit.

The DCO bot is incorrectly configured, otherwise nobody can interact with the branch directly from the web interface (like accepting proposals) or to merge `main` in the current branch when it's outdated.

**Review by everettraven — APPROVED** [review]
/lgtm

**Comment by everettraven** [comment]
> @everettraven can this be merged, or?

@dacamposol Yep, apologies for the delay!

**Pull request merged**
